oeps/oep-0017-feature-toggles.rst

```diff
@@ -0,0 +1,809 @@
++---------------------+----------------------------------------+------------------------+
+| Feature Category    | Optional field to group interdependent | In code, by developer  |
+|                     | toggles.                               |                        |
++---------------------+----------------------------------------+------------------------+
+| All Use Cases       | Lists one or more `Use Cases`_ to      | In code, by developer  |
+|                     | specify all expected usages of this    |                        |
+|                     | toggle.                                |                        |
++---------------------+----------------------------------------+------------------------+
+| Current Use Case(s) | A subset of "All Use Cases" to specify | In code, by developer; |
+|                     | the current state of this toggle.      | optionally editable via|
+|                     |                                        | admin interface.       |
++---------------------+----------------------------------------+------------------------+
+| Toggle Type         | One of Switch, Rollout, or Group to    | In code, by developer  |
+|                     | further clarify the toggle's usage.    |                        |
+|                     |                                        |                        |
++---------------------+----------------------------------------+------------------------+
+| Created in Code Date| Required field to specify the date the | In code, by developer  |
+|                     | toggle was added to the codebase; to   |                        |
+|                     | easily find all stale toggles.         |                        |
++---------------------+----------------------------------------+------------------------+
+| Expiration Date     | Optional field to specify target date  | In code, by developer  |
+|                     | of removal; to easily find all expired |                        |
+|                     | toggles.                               |                        |
++---------------------+----------------------------------------+------------------------+
+| First Modified Time | Date the toggle was first set in the   | Derived from relational|
+|                     | system; to get the starting date       | tables                 |
+|                     | of its use.                            |                        |
++---------------------+----------------------------------------+------------------------+
+| Last Modified Time  | Date the toggle was last set/unset in  | Derived from relational|
+|                     | the system; to easily find all unused  | tables                 |
+|                     | toggles.                               |                        |
++---------------------+----------------------------------------+------------------------+
```

Comment:
Thanks.
1. If one report per environment, I think it would be good for us to document somewhere what reports need to be reviewed (i.e. Prod, Edge).  Maybe that will always just be these two?  We had questions about other environments, and learned that they are managed through Open edX release process, so we just had to updated appropriate docs in those cases.
2. I can explain our use case for default value, and you can let me know if you have a better idea.
3. I'll explain soon the different values and you can let me know where this will appear.
Thanks!